test/shared_test/lib_shared_inverter_test.cpp

```diff
@@ -147,49 +147,49 @@ TEST_F(sharedInverterTest_lib_shared_inverter, tempDerateTest_lib_shared_inverte
 	V = 100.;
 	T = 9.;
 	reset();
-	inv->calculateTempDerate(V, T, pAC, eff, loss);
+	inv->calculateTempDerate(V, T, pAC, ratio, loss);
 	EXPECT_NEAR(pAC, 100, e) << "case 7";
 
 	// V<200 and 10 < T : extrapolated startT is 10. and slope is -0.1
 	V = 100.;
 	T = 11.;
 	reset();
-	inv->calculateTempDerate(V, T, pAC, eff, loss);
+	inv->calculateTempDerate(V, T, pAC, ratio, loss);
 	EXPECT_NEAR(pAC, 90, e) << "case 7";
 
 	// 200<V<300 and T < 25, interpolated startT is 25. and slope is -0.25
 	V = 250.;
 	T = 24.;
 	reset();
-	inv->calculateTempDerate(V, T, pAC, eff, loss);
+	inv->calculateTempDerate(V, T, pAC, ratio, loss);
 	EXPECT_NEAR(pAC, 100, e) << "case 9";
 
 	// 200<V<300 and 25 < T < 40, interpolated startT is 25. and slope is -0.25
 	V = 250.;
 	T = 26.;
 	reset();
-	inv->calculateTempDerate(V, T, pAC, eff, loss);
+	inv->calculateTempDerate(V, T, pAC, ratio, loss);
 	EXPECT_NEAR(pAC, 75, e) << "case 9";
 
 	// 200<V<300 and 40 < T, interpolated startT is 45. and slope is -0.45
 	V = 250.;
 	T = 46.;
 	reset();
-	inv->calculateTempDerate(V, T, pAC, eff, loss);
+	inv->calculateTempDerate(V, T, pAC, ratio, loss);
 	EXPECT_NEAR(pAC, 55, e) << "case 9";
 
 	// 300<V and T < 40, extrapolated startT is 40. and slope is -0.4
 	V = 400.;
 	T = 9.;
 	reset();
-	inv->calculateTempDerate(V, T, pAC, eff, loss);
+	inv->calculateTempDerate(V, T, pAC, ratio, loss);
 	EXPECT_NEAR(pAC, 100, e) << "case 9";
 
 	// 300<V and 40 < T, interpolated startT is 40. and slope is -0.4
 	V = 400.;
 	T = 41.;
 	reset();
-	inv->calculateTempDerate(V, T, pAC, eff, loss);
+	inv->calculateTempDerate(V, T, pAC, ratio, loss);
 	EXPECT_NEAR(pAC, 60, e) << "case 9";
 }
```

Comment:
Add a new test around `calculateACPower` similar to `calculateEffForACPower` but testing the temp derating instead.